src/arns.lua

```diff
@@ -499,7 +504,7 @@ function arns.assertValidBuyRecord(name, years, purchaseType, processId)
 
 	-- assert processId is valid pattern
 	assert(type(processId) == "string", "Process id is required and must be a string.")
-	assert(utils.isValidAOAddress(processId), "Process Id must be a valid AO signer address..")
+	assert(utils.isValidAddress(processId, true), "Process Id must be a valid address.")
```

Comment:
Nit: could make a `utils.isValidArNSTarget` function to consolidate the logic here.